lib/puppet/provider/zfs/zfs.rb

```diff
@@ -80,5 +80,21 @@ def exists?
     end
   end
 
+  # On FreeBSD zoned is called jailed
+  container = case Facter.value(:operatingsystem)
+    when "FreeBSD"
+      "jailed"
```

Comment:
The `define_method` calls below are expecting these to be symbols rather than strings. I think if you have this return symbols, the tests will start passing.